File: frontend/src/api/entity-types.ts
/**
 * Entity and Status types - must match backend models
 */

export const EntityType = {
  Goal: "goal",
  Capability: "capability",
  Limitation: "limitation",
  Aspiration: "aspiration",
} as const

export type EntityType = typeof EntityType[keyof typeof EntityType]

export const Status = {
  Pending: "pending",
  Progress: "progress",
  Completed: "completed",
  Abandoned: "abandoned",
} as const

export type Status = typeof Status[keyof typeof Status]

File: sdk/nowyouseeme/entity_types.py
"""
Entity and Status types - must match backend models
"""

from enum import Enum


class EntityType(str, Enum):
    """Entity type enumeration"""
    GOAL = "goal"
    CAPABILITY = "capability"
    LIMITATION = "limitation"
    ASPIRATION = "aspiration"


class Status(str, Enum):
    """Goal status enumeration"""
    PENDING = "pending"
    PROGRESS = "progress"
    COMPLETED = "completed"
    ABANDONED = "abandoned"
